tools/source_updater/updater/handler.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import subprocess
+from typing import Set
+
+from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from updater.catalog import CatalogMerger
+from updater.config import Config
+from updater.source import SourceRepository
+
+
+logger = logging.getLogger("handler")
+
+
+class SourceUpdaterHandler:
+    def __init__(self, source_repository: SourceRepository, catalog_merger: CatalogMerger):
+        self._source_repository = source_repository
+        self._catalog_merger = catalog_merger
+
+    def handle(self, source_name: str, new_manifest_source: ManifestDeclarativeSource, main_config: Config, other_configs: Set[Config]):
+        # FIXME eventually extract git related command somewhere it makes sense
+        # FIXME is there a state of the local git we would like to enforce?
+        branch_name = f"source-updater/updating-{source_name}"
+        validate_branch_process = subprocess.run(["git", "rev-parse", "--verify", branch_name])
+
+        if validate_branch_process.returncode == 0:
+            error_message = f"The target branch `{branch_name}` for the update operations already exist. Please make sure to push those " \
+                            f"local changes before doing more changes or delete this branch "
+            logger.error(error_message)
+            raise ValueError(error_message)
+
+        if not self._source_repository.exists(source_name):
+            error_message = "Source does not exist. Please generate it as demonstrated by " \
+                            "https://docs.airbyte.com/connector-development/config-based/tutorial/create-source "
+            raise ValueError(error_message)
+
+        catalog = self._source_repository.fetch_catalog(source_name)
+        self._catalog_merger.merge_and_save(source_name, catalog, new_manifest_source, main_config)
```

Comment:
I'm finding these interfaced to be a little bit convoluted. Right now the repository gets the catalog and passes it to the catalog_merger. And then the catalog merger uses the repository's update method. It just feels like these two objects are really dependent on each other's specific methods and implementation.

What if we were to refactor this a little bit so source_repository has a field catalog_merger. And the catalog_merger has no dependency on the repo at all. And instead of these two lines we just have one method `self._source_repository.update_catalog()`

And a rough flow of what the new`SourceRepository.update_catalog()` method look like:
1. Call to fetch the catalog (fetch_catalog)
2. Call the `catalog_merger.merge()` to perform the catalog transformation to add/remove streams (but does not save it).
3. Call to update the catalog 

Does that make sense? I was just thinking about this from the angle of how we can simplify how we invoke fetch + save behind a simpler interface.

Reply:
I think it makes a lot of sense and I'm glad you shared this with me. I've simplified this, thanks!